File: JLessons/JL3.java
//JL3==exceptions in java
import java.util.Date;
import java.lang.Exception;

public class JL3
{
    public static void main(String[] args) {
        Date d=null;
        try {
            System.out.println(d.getTime());
        } catch (Exception e) {
            
            System.out.println(e.toString());
        }
        int a=0;
        int b=0;
        try {
            a=a/b;
        } catch (Exception e) {
            
            System.out.println(e.toString());
        }
        try {
            throw new MyE();
        } catch (Exception e) {
            
            System.out.println(e.toString());
        }
        double c=0;
        double d1=0;
        c=c/d1;
        System.out.println("c="+c);


        
    }
}

/**
 * MyE
 */
class MyE extends Exception{

@Override
public String toString() {
    
    return "WDNMD\n";
}   
}

File: JLessons/JL3.cpp
//JL3==exceptions in java
#include <exception>
#include <stdexcept>
#include <iostream>
#include <csignal>//for signal
#include <csetjmp>//for longjmp


class MyE : public std::exception
{
private:
    /* data */
public:

    const char* what()const throw()  override{return "wdnmd\n";}
    MyE()=default;


};
std::jmp_buf jmpenv;
void FPE_Handler(int signal)
{
    std::cerr<<"caught SIGFPE\("<<signal<<"\)\n";
    std::longjmp(jmpenv,signal);
    //throw MyE();
     //siginterrupt(SIGFPE, 1);//https://www.mkssoftware.com/docs/man3/siginterrupt.3.asp  https://stackoverflow.com/a/14906044
}




int main(int argc, char const *argv[])
{
    try{

        throw MyE();
    }
    catch(std::exception& e){

        std::cerr<<e.what();
    }
 double c=0;
 double d=0;
 c=c/d;
 std::cerr<<"c="<<c<<'\n';
std::signal(SIGFPE,FPE_Handler); 
 //std::signal(SIGFPE,SIG_IGN);

 int a=0;
 int b=0;
if(!setjmp(jmpenv))
a=a/b;


 

 //std::raise(SIGFPE);
 std::cout<<"all exception tests done\n";


    return 0;
}
